src/lightning/fabric/utilities/spike.py

```diff
@@ -0,0 +1,165 @@
+import json
+import operator
+import os
+import warnings
+from typing import Any, Dict, List, Literal, Optional, TYPE_CHECKING, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version
+
+_TORCHMETRICS_GREATER_EQUAL_1_0_0 = compare_version("torchmetrics", operator.ge, "1.0.0")
+
+if TYPE_CHECKING:
+    from lightning.fabric.fabric import Fabric
+
+
+class SpikeDetection:
+    """Spike Detection Callback.
+
+    Terminates training with a ``TrainingSpikeException`` when a loss-spike was detected and
+    saves the batches to skip when resuming to a file.
+
+    We skip the current and the previous batch since it is unclear whether the previous batch
+    altered the weights in a way that it causes the spike or just the current batch is corrupted somehow.
+
+    Args:
+        mode: Whether to minimize or maximize the tracked metric
+        window: A running mean of metrics with ``window`` size. Serves as reference value for spikes.
+        warmup: After how many batches spike-tracking should start
+        atol: An absolute tolerance.  Every diff between the running mean and the current value,
+            that's not an improvement and above ``atol`` will be considered a spike
+        rtol: A relative tolerance. Every diff between the running mean and the current value,
+            that's higher than ``rtol * running_mean`` is considered a spike
+        exclude_batches_path: Where to save the file that contains the batches to exclude. Will default to current directory.
+        finite_only: If set to ``False``, consider non-finite values like NaN, inf and -inf a spike as well.
+    """
+
+    def __init__(
+        self,
+        mode: Literal["min", "max"] = "min",
+        window: int = 10,
+        warmup: int = 1,
+        atol: Optional[float] = None,
+        rtol: Optional[float] = 2.0,
+        exclude_batches_path: Optional[str] = None,
+        finite_only: bool = True,
+    ):
+        if _TORCHMETRICS_GREATER_EQUAL_1_0_0:
+            from torchmetrics.aggregation import MeanMetric
+            from torchmetrics.wrappers import Running
+        else:
+            raise RuntimeError("SpikeDetection requires torchmetrics>=1.0.0! Please upgrade your version!")
+        super().__init__()
+
+        self.last_val: Union[torch.Tensor, float] = 0.0
```

Comment:
Shouldn't this be just tensor? I don't see where it could ever be just a float. Maybe standardize to just tensor to simplify the conversion in state_dict method.

Reply:
it's a float here in the initialization because this way I don't need to move it to the correct device :D

Reply:
In this case I would standardize to just float, but this is a nitpick. Feel free to resolve.